src/Native/MatrixFactorizationNative/CMakeLists.txt

```diff
@@ -1,16 +1,32 @@
 project (MatrixFactorizationNative)
 add_definitions(-D_SCL_SECURE_NO_WARNINGS)
 add_definitions(-DUSEOMP)
-add_definitions(-DUSESSE)
+
+if(NOT ${ARCHITECTURE} MATCHES "arm.*")
+    add_definitions(-DUSESSE)
+endif()
 
 include_directories(libmf)
 
 if(UNIX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -O3 -pthread -std=c++0x -march=native")
+    if(NOT ${ARCHITECTURE} MATCHES "arm.*")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -O3 -pthread -std=c++0x -march=native")
+    else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -O3 -pthread -std=c++0x")
```

Comment:
Can we factor this so it isn't duplicated?

Reply:
Refactored.